### PR TITLE
Update helm/kubetl binary and remove locksmith dep

### PR DESCRIPTION
**Pull request by mate4st**
**What this PR does / why we need it**:

Bump Helm and Kubelet in Dockerfile.. 
Removes archived `github.com/coreos/locksmith` dependency. 

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
Please describe your PR. Like, this is not just bumping Helm, you also introduced an ENUM Into our API. That warrants some words of description and context.

**Comment by WeirdMachine** [comment]
> Please describe your PR. Like, this is not just bumping Helm, you also introduced an ENUM Into our API. That warrants some words of description and context.

I had to remove api validation again as we allow different time layouts it couldn't be handled without code

File: pkg/apis/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -360,6 +360,7 @@ type ClusterConditionType string
 // The reference time for this is the node system time and might differ from
 // the user's timezone, which needs to be considered when configuring a window.
 type UpdateWindow struct {
+
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go:363]: unneccessary change

**Review by xrstf — APPROVED** [review]
/approve

please stop adding random newlines everywhere in the future ;-)

**Pull request merged**